src/odemis/gui/cont/views.py

```diff
@@ -285,22 +290,74 @@ def focusViewWithStream(self, stream):
 
         logging.debug("Failed to find any view compatible with stream %s", stream.name.value)
 
+OVV_DIMENSION = (0.02, 0.02)  # m
 
 class OverviewController(object):
-    """ Small class to connect stage history and overview canvas together """
-
-    def __init__(self, tab_data, overview_canvas):
+    """ Class to connect stage history and overview canvas together and to control the overview image  """
 
+    def __init__(self, main_data, tab_data, overview_canvas, m_view):
         self._data_model = tab_data
-        self.overview_canvas = overview_canvas
+        self.canvas = overview_canvas
+        self.m_view = m_view
+        self.main_data = main_data
+
+        self.curr_s = None
 
         if tab_data.main.stage:
             tab_data.main.stage.position.subscribe(self.on_stage_pos_change, init=True)
             tab_data.main.chamberState.subscribe(self._on_chamber_state)
+            tab_data.streams.subscribe(self._on_current_stream)
+
+        # Global overview image (Delphi)
+        overview_stream = None
+        if main_data.overview_ccd:
+            # Overview camera can be RGB => in that case len(shape) == 4
+            if len(main_data.overview_ccd.shape) == 4:
+                overview_stream = acqstream.RGBCameraStream("Overview", main_data.overview_ccd,
+                                                            main_data.overview_ccd.data, None)
+            else:
+                overview_stream = acqstream.BrightfieldStream("Overview", main_data.overview_ccd,
+                                                              main_data.overview_ccd.data, None)
+            self.m_view.addStream(overview_stream)
+            # TODO: add it to self.tab_data_model.streams?
+
+        # Built-up overview image
+        # Initialize the size of the ovv image with the stage size if the stage is small (< 5cm),
+        # otherwise fall back to OVV_DIMENSION
+        ax_x = self.main_data.stage.axes["x"]
+        ax_y = self.main_data.stage.axes["y"]
+        if (hasattr(ax_x, "range") and hasattr(ax_y, "range")):
+            max_x = max(numpy.absolute(ax_x.range))
```

Comment:
The range can be something like [1.1, 1.2], in which case it'll have a 0.1m "range width". You actually only want to know the width of the range. => use `ax_x[1] - ax_x[0]`.